mithril-common/src/messages/message_parts/signer.rs

```diff
@@ -73,7 +73,9 @@ impl SignerWithStakeMessagePart {
                 verification_key: message.verification_key.try_into()?,
                 verification_key_signature: message.verification_key_signature,
                 kes_period: message.kes_period,
-                operational_certificate: message.operational_certificate,
+                operational_certificate: message
+                    .operational_certificate
+                    .and_then(|op_cert| (op_cert.try_into().ok())),
```

Comment:
Using `.ok()` hide eventual errors, you should propagate the error instead using a `?`. Sadly you can't use `?` inside closures so you will need to convert this to a `match`.